code/modules/reagents/Chemistry-Machinery.dm

```diff
@@ -1300,14 +1301,15 @@ obj/machinery/computer/pandemic/proc/replicator_cooldown(var/waittime)
 	var/state_change = 0
 	if(on)
 		if(beaker)
-			if(beaker.reagents.chem_temp > temperature)
-				beaker.reagents.chem_temp = max(beaker.reagents.chem_temp-rate, temperature)
-				beaker.reagents.handle_reactions()
-				state_change = 1
-			else if(beaker.reagents.chem_temp < temperature)
-				beaker.reagents.chem_temp = min(beaker.reagents.chem_temp+rate, temperature)
+			if(beaker.reagents.chem_temp != desired_temp)
+				beaker.reagents.chem_temp += ((desired_temp - beaker.reagents.chem_temp) * heater_coefficient)
+
+				beaker.reagents.chem_temp = round(beaker.reagents.chem_temp) //stops stuff like 456.12312312302
+				if(beaker.reagents.chem_temp == ((desired_temp-3) || (desired_temp+3)))//need better way to stop superdecimals and rounding drops

```

Comment:
You might actually want to introduce some error, possibly scalable by the components used to make the heater. That way, a heater kitbashed by an assistant from Space Chinese parts and rust could be so badly thermostated you could actually trigger a heat-based reaction that blows the chem up in your face.